examples/src/main/scala/org/apache/spark/examples/ml/GaussianMixtureExample.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.examples.ml
+
+// scalastyle:off println
+
+import org.apache.spark.{SparkConf, SparkContext}
+// $example on$
+import org.apache.spark.ml.clustering.{GaussianMixture, GaussianMixtureModel}
+import org.apache.spark.mllib.linalg.{Vectors, VectorUDT}
+import org.apache.spark.sql.{DataFrame, Row, SparkSession}
+import org.apache.spark.sql.types.{StructField, StructType}
+// $example off$
+
+/**
+ * An example demonstrating Gaussian Mixture Model (GMM).
+ * Run with
+ * {{{
+ * bin/run-example ml.GaussianMixtureExample
+ * }}}
+ */
+object GaussianMixtureExample {
+  def main(args: Array[String]): Unit = {
+    // Creates a SparkSession
+    val spark = SparkSession.builder.appName(s"${this.getClass.getSimpleName}").getOrCreate()
+    val input = "data/mllib/gmm_data.txt"
+
+    // $example on$
+    // Crates a DataFrame
+    val rowRDD = spark.read.text(input).rdd.filter(_.nonEmpty)
```

Comment:
Agree, using the same kmeans libsvm data is fine. It's just for
illustrative purposes.
On Mon, 9 May 2016 at 20:12, Seth Hendrickson <notifications@github.com>
wrote:

> In
> examples/src/main/scala/org/apache/spark/examples/ml/GaussianMixtureExample.scala
> <https://github.com/apache/spark/pull/12788#discussion_r62545534>:
>
> > +/**
> > + * An example demonstrating Gaussian Mixture Model (GMM).
> > + * Run with
> > + * {{{
> > + * bin/run-example ml.GaussianMixtureExample
> > + * }}}
> > + */
> > +object GaussianMixtureExample {
> > +  def main(args: Array[String]): Unit = {
> > +    // Creates a SparkSession
> > +    val spark = SparkSession.builder.appName(s"${this.getClass.getSimpleName}").getOrCreate()
> > +    val input = "data/mllib/gmm_data.txt"
> > +
> > +    // $example on$
> > +    // Crates a DataFrame
> > +    val rowRDD = spark.read.text(input).rdd.filter(_.nonEmpty)
>
> I don't think it's necessary, to be honest. We could change the name of
> the kmeans data to be something like clustering_data.txt, but I think
> it's fine to just use the kmeans file here. It would be a bit excessive to
> create two separate files with exactly the same data to me. @zhengruifeng
> <https://github.com/zhengruifeng>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/12788/files/4b5152a43062f382983ecb8084536cdd2c3849d2#r62545534>
>